### PR TITLE
envoy: removing srds from E-M

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/filters/network/http_connection_manager/config.cc [hunk 1]
```diff
@@ -522,6 +526,8 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     early_header_mutation_extensions_.push_back(std::move(extension));
   }
 
+  auto srds_factory =
+      Envoy::Config::Utility::getFactoryByName<Router::SrdsFactory>(srds_factory_name);
```
Comment (adisuissa) [inline source/extensions/filters/network/http_connection_manager/config.cc:530]: nit: move to the kScopedRoutes switch case.

File: envoy/config/config_provider_manager.h [hunk 2]
```diff
@@ -23,7 +24,7 @@ namespace Config {
  * growth based on the size of the configuration set, regardless of the number of threads/objects
  * that must hold a reference/pointer to them.
  */
-class ConfigProviderManager {
+class ConfigProviderManager : public Singleton::Instance {
```
Comment (adisuissa) [inline envoy/config/config_provider_manager.h:27]: I can see why one would like to have a ConfigProviderManager as singleton, but it is unclear to me what triggered the change moving the Singleton inheritance from the implementation to the interface?
(I'm assuming this is something related to C++'s unique_ptr not aligned well with inheritance, but just want to make sure this change is needed).

**Review by adisuissa — COMMENTED** [review]
Overall LGTM, thanks!
I don't think we have technical release notes, but if there's something like that, should notify SRDS users that they need to link the "//source/common/router:scoped_rds_lib" or otherwise it will fail during runtime.

File: source/extensions/filters/network/http_connection_manager/config.cc (hunk 1 above)
Comment (alyssawilk) [reply source/extensions/filters/network/http_connection_manager/config.cc:530]: mind if I leave here? it's a pretty cheap function for the non-srds case and if you declare variables in a switch  you need braces and it screws up coverage.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Review by zuercher — APPROVED** [review]

**Pull request merged**